modules/drivers/oracle/src/metabase/driver/oracle.clj

```diff
@@ -599,22 +599,29 @@
   (fn []
     (.getString rs i)))
 
+(defn- offset-date-time-value
+  "Returns the `OffsetDateTime` value of the `TIMESTAMPTZ` or `TIMESTAMPLTZ` object `timestamp`."
+  [^ResultSet rs timestamp]
+  ;; Oracle `TIMESTAMPTZ` or `TIMESTAMPLTZ` types can have either a zone offset *or* a zone ID; you could fetch either
+  ;; `OffsetDateTime` or `ZonedDateTime` using `.getObject`, but fetching the wrong type will result in an Exception,
+  ;; meaning we have try both and wrap the first in a try-catch. As far as I know there's no way to tell whether the
+  ;; value has a zone offset or ID without first fetching a `TIMESTAMPTZ` object. So to avoid the try-catch we can
+  ;; fetch the `TIMESTAMPTZ` or `TIMESTAMPLTZ` and use `.timestampValue` instead.
+  (let [^C3P0ProxyConnection proxy-conn (.. rs getStatement getConnection)
+        conn                            (.unwrap proxy-conn OracleConnection)]
+    (t/offset-date-time (.timestampValue timestamp conn) (t/zone-offset 0))))
+
 (defmethod sql-jdbc.execute/read-column-thunk [:oracle OracleTypes/TIMESTAMPTZ]
   [_driver ^ResultSet rs _rsmeta ^Integer i]
-  ;; Oracle `TIMESTAMPTZ` types can have either a zone offset *or* a zone ID; you could fetch either `OffsetDateTime`
-  ;; or `ZonedDateTime` using `.getObject`, but fetching the wrong type will result in an Exception, meaning we have
-  ;; try both and wrap the first in a try-catch. As far as I know there's now way to tell whether the value has a zone
-  ;; offset or ID without first fetching a `TIMESTAMPTZ` object. So to avoid the try-catch we can fetch the
-  ;; `TIMESTAMPTZ` and use `.offsetDateTimeValue` instead.
   (fn []
     (when-let [^TIMESTAMPTZ t (.getObject rs i TIMESTAMPTZ)]
-      (let [^C3P0ProxyConnection proxy-conn (.. rs getStatement getConnection)
-            conn                            (.unwrap proxy-conn OracleConnection)]
-        ;; TIMEZONE FIXME - we need to warn if the Oracle JDBC driver is `ojdbc7.jar`, which probably won't have this
-        ;; method
-        ;;
-        ;; I think we can call `(oracle.jdbc.OracleDriver/getJDBCVersion)` and check whether it returns 4.2+
```

Comment:
This comment isn't necessary anymore because Java 7 has been "end of life" since April 2015, and additionally Metabase only supports version 19c, which isn't compatible with `odjbc7.jar`. 


![image](https://github.com/metabase/metabase/assets/39073188/bb9ef81c-8d16-40d2-805d-ee26e3e51967)
([source](https://www.oracle.com/database/technologies/faq-jdbc.html))

Reply:
Good spotting. It shouldn't really matter because we're not officially supporting that version, but according to the docs ojdbc6.jar has `.timestampValue` with two args: https://javadoc.io/doc/com.oracle.database.jdbc/ojdbc6/latest/index.html

Reply:
We've implicitly required `ojdbc8.jar` since at least 2020 now since we've had that call to `.offsetDateTimeValue`.

I think that table you screenshotted above is just listing the driver versions they released along with releasing specific database versions.

This table shows that those release versions all work with 11.2 -- the three releases of `ojdbc8.jar` all work with 11.2
![image](https://github.com/metabase/metabase/assets/1455846/733163cb-204f-46be-8a84-21ebd0b77160)